File: fe2-android/app/src/test/java/com/github/dedis/popstellar/model/network/method/MessageTest.java
package com.github.dedis.popstellar.model.network.method;

import com.github.dedis.popstellar.model.objects.Channel;

import org.junit.Test;

import java.util.Objects;

import static org.junit.Assert.*;

public class MessageTest {
  private static final Channel CHANNEL = Channel.fromString("root/foo");
  private static final Message MESSAGE =
      new Message(CHANNEL) {
        @Override
        public String getMethod() {
          return "bar";
        }
      };

  @Test
  public void testEquals() {
    assertEquals(MESSAGE, MESSAGE);
    assertNotEquals(null, MESSAGE);

    Message message2 =
        new Message(Channel.fromString("some channel")) {
          @Override
          public String getMethod() {
            return "bar";
          }
        };
    assertNotEquals(MESSAGE, message2);
  }

  @Test
  public void testHashCode() {
    assertEquals(Objects.hash(CHANNEL), MESSAGE.hashCode());
  }

  @Test
  public void constructorThrowsExceptionForNullChannel() {
    assertThrows(
        IllegalArgumentException.class,
        () ->
            new Message(null) {
              @Override
              public String getMethod() {
                return "bar";
              }
            });
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/MessageTest.kt
package com.github.dedis.popstellar.model.network.method

import com.github.dedis.popstellar.model.objects.Channel.Companion.fromString
import java.util.Objects
import org.junit.Assert
import org.junit.Test

class MessageTest {
  @Test
  fun testEquals() {
    Assert.assertEquals(MESSAGE, MESSAGE)
    Assert.assertNotEquals(null, MESSAGE)

    val message2: Message =
      object : Message(fromString("some channel")) {
        override val method: String
          get() = "bar"
      }
    Assert.assertNotEquals(MESSAGE, message2)
  }

  @Test
  fun testHashCode() {
    Assert.assertEquals(Objects.hash(CHANNEL).toLong(), MESSAGE.hashCode().toLong())
  }

  @Test
  fun constructorThrowsExceptionForNullChannel() {
    Assert.assertThrows(IllegalArgumentException::class.java) {
      object : Message(null) {
        override val method: String
          get() = "bar"
      }
    }
  }

  companion object {
    private val CHANNEL = fromString("root/foo")
    private val MESSAGE: Message =
      object : Message(CHANNEL) {
        override val method: String
          get() = "bar"
      }
  }
}
